build-system/common/colors.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const colors = require('kleur/colors');
+
+/**@type {Record<keyof typeof colors, (input?: string | boolean | number | string[] | number[] | boolean []) => string>}   */
+module.exports = Object.entries(colors).reduce((map, [key, formatter]) => {
+  if (key === 'enabled') {
+    map[key] = formatter;
+    return map;
+  }
+  map[key] = function (input) {
+    if (!input) {
+      return `${input}`;
+    }
+
+    if (input instanceof Array) {
+      return /** @type {function} */ (formatter)(input.join(' '));
+    }
+
+    return /** @type {function} */ (formatter)(input);
+  };
+  return map;
+}, /**@type {Record<keyof typeof colors, (input?: string | boolean | number | string[] | number[] | boolean []) => string>}   */ ({}));
```

Comment:
While this function addresses all type errors discovered by `amp check-build-system`, I suspect it is doing too much. Things like error checking, handling `null` / `undefined`, formatting of arrays / objects, etc. should be left to the underlying library. Doing it here could prove brittle when future updates to the library introduce subtle changes.

I played around with this code on my local branch and came up with a more concise alternative:

```suggestion
/**
 * @fileoverview The kleur/color library provides a variety of untyped color
 * formatting functions. This file provides a typed wrapper for each of them in
 * order to satisfy our type checks without changing any functionality. For more
 * info, see https://github.com/lukeed/kleur/blob/master/colors.{mjs,d.ts}.
 */

module.exports = Object.entries(colors).reduce((map, [key, formatter]) => {
  map[key] =
    typeof formatter == 'function'
      ? (txt) => /** @type {function} */ (formatter)(txt)
      : formatter;
  return map;
}, /**@type {Record<keyof typeof colors, (input?: *) => string>} */ ({}));
```

With this, all these invocations were free of type errors (no red squiggly lines):

![image](https://user-images.githubusercontent.com/26553114/117499760-a9224180-af49-11eb-92c5-f5f37048f4f7.png)


In addition, there were zero errors returned by `amp check-build-system` due to `colors.js`.

WDYT?